test/fixtures/hdfs-fixture/build.gradle

```diff
@@ -18,25 +18,22 @@
  */
 
 apply plugin: 'elasticsearch.build'
+apply plugin: 'elasticsearch.test.fixtures'
 
-versions << [
-  'hadoop2': '2.8.1'
-]
-
-// we create MiniHdfsCluster with the hadoop artifact
 dependencies {
-  compile "org.apache.hadoop:hadoop-minicluster:${versions.hadoop2}"
+  compile "org.apache.hadoop:hadoop-minicluster:2.8.1"
+}
+
+task syncClasses(type: Sync) {
+  from sourceSets.test.compileClasspath
+  from sourceSets.test.output.dirs
```

Comment:
This can just be `sourceSets.test.output`. This also has the advantage that it returns a `Buildable` so you can ditch the `dependsOn` for the compile task on line 31.